File: mobile/ios/shared/OrbitDesignSystem.swift
import SwiftUI

// ─── Orbit Design System Specifications ───────────────────────────────────────

public struct OrbitDesignSystem {
    public static let orbitCyan = Color(red: 0.0, green: 0.75, blue: 1.0)
    public static let orbitPurple = Color(red: 0.63, green: 0.13, blue: 0.94)
    public static let spaceNavy = Color.black
    public static let spaceNavyLight = Color(red: 0.04, green: 0.04, blue: 0.04)
    public static let spaceNavyLighter = Color(red: 0.07, green: 0.07, blue: 0.07)
    public static let orbitBorder = Color(red: 0.10, green: 0.10, blue: 0.18)
    public static let mutedText = Color(red: 0.49, green: 0.72, blue: 0.88)
    public static let destructive = Color(red: 1.0, green: 0.27, blue: 0.27)
    
    public static let orbitGradient = LinearGradient(
        gradient: Gradient(colors: [orbitCyan, orbitPurple]),
        startPoint: .leading,
        endPoint: .trailing
    )
}

// ─── 1. Orbit Custom Buttons ──────────────────────────────────────────────────

public struct OrbitGradientButton: View {
    public let text: String
    public let onClick: () -> Void
    public var enabled: Bool = true
    
    public init(text: String, onClick: @escaping () -> Void, enabled: Bool = true) {
        self.text = text
        self.onClick = onClick
        self.enabled = enabled
    }
    
    public var body: some View {
        Button(action: onClick) {
            Text(text)
                .font(.headline)
                .bold()
                .foregroundColor(.white)
                .frame(maxWidth: .infinity)
                .frame(height: 50)
                .background(enabled ? OrbitDesignSystem.orbitGradient : LinearGradient(gradient: Gradient(colors: [Color.gray, Color.darkGray]), startPoint: .leading, endPoint: .trailing))
                .cornerRadius(12)
        }
        .disabled(!enabled)
    }
}

// ─── 2. Orbit Glass Cards ─────────────────────────────────────────────────────

public struct OrbitGlassCard<Content: View>: View {
    public let borderColor: Color
    public let content: () -> Content
    
    public init(borderColor: Color = OrbitDesignSystem.orbitBorder, @ViewBuilder content: @escaping () -> Content) {
        self.borderColor = borderColor
        self.content = content
    }
    
    public var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            content()
        }
        .padding(20)
        .background(OrbitDesignSystem.spaceNavyLight)
        .cornerRadius(16)
        .overlay(RoundedRectangle(cornerRadius: 16).stroke(borderColor, lineWidth: 1))
    }
}

// ─── 3. Orbit Text Fields & Search Bars ───────────────────────────────────────

public struct OrbitInput: View {
    public let label: String
    @Binding public var text: String
    
    public init(label: String, text: Binding<String>) {
        self.label = label
        self._text = text
    }
    
    public var body: some View {
        TextField(label, text: $text)
            .padding()
            .background(OrbitDesignSystem.spaceNavyLighter)
            .cornerRadius(10)
            .foregroundColor(.white)
            .overlay(RoundedRectangle(cornerRadius: 10).stroke(OrbitDesignSystem.orbitBorder, lineWidth: 1))
    }
}

public struct OrbitSearchBar: View {
    @Binding public var query: String
    
    public init(query: Binding<String>) {
        self._query = query
    }
    
    public var body: some View {
        HStack {
            Image(systemName: "magnifyingglass")
                .foregroundColor(OrbitDesignSystem.orbitCyan)
            TextField("Search Orbit catalog...", text: $query)
                .foregroundColor(.white)
        }
        .padding()
        .background(OrbitDesignSystem.spaceNavyLight)
        .cornerRadius(24)
        .overlay(RoundedRectangle(cornerRadius: 24).stroke(OrbitDesignSystem.orbitBorder, lineWidth: 1))
    }
}

// ─── 4. Orbit Top Bars & Headers ──────────────────────────────────────────────

public struct OrbitTopBar: View {
    public let title: String
    public var subtitle: String? = nil
    
    public init(title: String, subtitle: String? = nil) {
        self.title = title
        self.subtitle = subtitle
    }
    
    public var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(title)
                .font(.title2)
                .bold()
                .foregroundColor(.white)
            if let subtitle = subtitle {
                Text(subtitle)
                    .font(.subheadline)
                    .foregroundColor(OrbitDesignSystem.mutedText)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(.bottom, 16)
    }
}

// ─── 5. Orbit Specialized Cards ───────────────────────────────────────────────

public struct OrbitPackageCard: View {
    public let title: String
    public let subtitle: String
    public let price: String
    public let features: [String]
    public let onSelect: () -> Void
    public var isPopular: Bool = false
    
    public var body: some View {
        OrbitGlassCard(borderColor: isPopular ? OrbitDesignSystem.orbitPurple : OrbitDesignSystem.orbitBorder) {
            HStack {
                VStack(alignment: .leading) {
                    Text(title)
                        .font(.title3)
                        .bold()
                        .foregroundColor(.white)
                    if isPopular {
                        Text("MOST POPULAR")
                            .font(.caption2)
                            .bold()
                            .padding(.horizontal, 6)
                            .padding(.vertical, 2)
                            .background(OrbitDesignSystem.orbitPurple.opacity(0.2))
                            .foregroundColor(OrbitDesignSystem.orbitPurple)
                            .cornerRadius(4)
                    }
                }
                Spacer()
                Text(price)
                    .font(.title2)
                    .bold()
                    .foregroundColor(isPopular ? OrbitDesignSystem.orbitPurple : OrbitDesignSystem.orbitCyan)
            }
            
            Text(subtitle)
                .font(.caption)
                .foregroundColor(OrbitDesignSystem.mutedText)
                .padding(.vertical, 8)
            
            VStack(alignment: .leading, spacing: 6) {
                ForEach(features, id: \.self) { feature in
                    Text("✓ \(feature)")
                        .font(.subheadline)
                        .foregroundColor(.white)
                }
            }
            
            Spacer().frame(height: 12)
            OrbitGradientButton(text: "Book \(title) Package", onClick: onSelect)
        }
    }
}

public struct OrbitStatusTimeline: View {
    public let currentStage: Int
    public let stages: [String]
    
    public var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            ForEach(Array(stages.enumerated()), id: \.offset) { index, stage in
                HStack {
                    if index < currentStage {
                        Text("✓").bold().foregroundColor(OrbitDesignSystem.orbitCyan)
                        Text(stage).foregroundColor(.white)
                    } else if index == currentStage {
                        Text("•").bold().foregroundColor(OrbitDesignSystem.orbitCyan)
                        Text("\(stage) (Active)").bold().foregroundColor(OrbitDesignSystem.orbitCyan)
                    } else {
                        Text("◦").foregroundColor(OrbitDesignSystem.mutedText)
                        Text("\(stage) (Pending)").foregroundColor(OrbitDesignSystem.mutedText)
                    }
                }
            }
        }
    }
}

public struct OrbitWalletCard: View {
    public let balance: String
    public let pending: String
    public let onWithdraw: () -> Void
    
    public var body: some View {
        OrbitGlassCard(borderColor: OrbitDesignSystem.orbitCyan) {
            Text("Available Balance")
                .font(.caption)
                .foregroundColor(OrbitDesignSystem.mutedText)
            Text(balance)
                .font(.system(size: 36, weight: .black))
                .foregroundColor(.white)
            Text("Pending Clearance: \(pending)")
                .font(.caption)
                .bold()
                .foregroundColor(OrbitDesignSystem.orbitCyan)
            
            Spacer().frame(height: 12)
            OrbitGradientButton(text: "Initiate Instant Payout", onClick: onWithdraw)
        }
    }
}

public struct OrbitLoader: View {
    public var body: some View {
        ProgressView()
            .tint(OrbitDesignSystem.orbitCyan)
            .scaleEffect(1.5)
    }
}

public struct OrbitSkeleton: View {
    public var body: some View {
        RoundedRectangle(cornerRadius: 16)
            .fill(OrbitDesignSystem.spaceNavyLight)
            .frame(height: 100)
            .overlay(RoundedRectangle(cornerRadius: 16).stroke(OrbitDesignSystem.orbitBorder, lineWidth: 1))
    }
}

File: mobile/android/core/src/main/java/com/orbitlogic/core/ui/OrbitDesignSystem.kt
package com.orbitlogic.core.ui

import androidx.compose.animation.*
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

// ─── Orbit Brand Tokens ───────────────────────────────────────────────────────

val OrbitCyan = Color(0xFF00BFFF)
val OrbitPurple = Color(0xFFA020F0)
val SpaceNavy = Color(0xFF000000)
val SpaceNavyLight = Color(0xFF0A0A0A)
val SpaceNavyLighter = Color(0xFF111111)
val OrbitBorder = Color(0xFF1A1A2E)
val MutedText = Color(0xFF7EB8E0)
val Destructive = Color(0xFFFF4444)

val OrbitGradientBrush = Brush.linearGradient(listOf(OrbitCyan, OrbitPurple))
val OrbitPurpleCyanBrush = Brush.linearGradient(listOf(OrbitPurple, OrbitCyan))

// ─── 1. Orbit Buttons ─────────────────────────────────────────────────────────

@Composable
fun OrbitGradientButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true
) {
    Button(
        onClick = onClick,
        enabled = enabled,
        colors = ButtonDefaults.buttonColors(containerColor = Color.Transparent),
        contentPadding = PaddingValues(),
        shape = RoundedCornerShape(12.dp),
        modifier = modifier.height(50.dp)
    ) {
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(
                    if (enabled) OrbitGradientBrush
                    else Brush.linearGradient(listOf(Color.DarkGray, Color.Gray))
                ),
            contentAlignment = Alignment.Center
        ) {
            Text(text = text, color = Color.White, fontWeight = FontWeight.Bold, fontSize = 16.sp)
        }
    }
}

// ─── 2. Orbit Cards (Glassmorphism on Black) ───────────────────────────────

@Composable
fun OrbitGlassCard(
    modifier: Modifier = Modifier,
    borderColor: Color = OrbitBorder,
    content: @Composable ColumnScope.() -> Unit
) {
    Card(
        colors = CardDefaults.cardColors(containerColor = SpaceNavyLight),
        shape = RoundedCornerShape(16.dp),
        modifier = modifier
            .fillMaxWidth()
            .border(1.dp, borderColor, RoundedCornerShape(16.dp))
    ) {
        Column(
            modifier = Modifier.padding(20.dp),
            content = content
        )
    }
}

@Composable
fun OrbitCardStrong(
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit
) {
    Card(
        colors = CardDefaults.cardColors(containerColor = SpaceNavyLighter),
        shape = RoundedCornerShape(16.dp),
        modifier = modifier
            .fillMaxWidth()
            .border(1.dp, OrbitCyan.copy(alpha = 0.3f), RoundedCornerShape(16.dp))
    ) {
        Column(
            modifier = Modifier.padding(20.dp),
            content = content
        )
    }
}

// ─── 3. Orbit Text Fields & Search Inputs ───────────────────────────────────

@Composable
fun OrbitInput(
    value: String,
    onValueChange: (String) -> Unit,
    label: String,
    modifier: Modifier = Modifier,
    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
    singleLine: Boolean = true,
    minLines: Int = 1
) {
    OutlinedTextField(
        value = value,
        onValueChange = onValueChange,
        label = { Text(label) },
        modifier = modifier.fillMaxWidth(),
        singleLine = singleLine,
        minLines = minLines,
        keyboardOptions = keyboardOptions,
        colors = OutlinedTextFieldDefaults.colors(
            focusedBorderColor = OrbitCyan,
            unfocusedBorderColor = OrbitBorder,
            focusedLabelColor = OrbitCyan,
            unfocusedLabelColor = MutedText
        )
    )
}

@Composable
fun OrbitSearchBar(
    query: String,
    onQueryChange: (String) -> Unit,
    modifier: Modifier = Modifier
) {
    OutlinedTextField(
        value = query,
        onValueChange = onQueryChange,
        placeholder = { Text("Search Orbit catalog...", color = MutedText) },
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(24.dp)),
        singleLine = true,
        colors = OutlinedTextFieldDefaults.colors(
            focusedBorderColor = OrbitCyan,
            unfocusedBorderColor = OrbitBorder,
            containerColor = SpaceNavyLight
        )
    )
}

// ─── 4. Orbit Headers & Top Bars ──────────────────────────────────────────────

@Composable
fun OrbitTopBar(title: String, subtitle: String? = null) {
    Column(modifier = Modifier.fillMaxWidth().padding(bottom = 16.dp)) {
        Text(
            text = title,
            fontSize = 24.sp,
            fontWeight = FontWeight.Bold,
            color = Color.White
        )
        subtitle?.let {
            Text(
                text = it,
                fontSize = 14.sp,
                color = MutedText,
                modifier = Modifier.padding(top = 2.dp)
            )
        }
    }
}

// ─── 5. Orbit Navigation Pill (`orbit-nav-pill`) ─────────────────────────────

@Composable
fun OrbitBottomNavigation(
    tabs: List<Pair<String, String>>,
    currentTab: String,
    onSelectTab: (String) -> Unit
) {
    Surface(
        color = SpaceNavyLight,
        tonalElevation = 12.dp,
        modifier = Modifier
            .fillMaxWidth()
            .height(64.dp)
            .border(1.dp, OrbitBorder)
    ) {
        Row(
            modifier = Modifier.fillMaxSize(),
            horizontalArrangement = Arrangement.SpaceAround,
            verticalAlignment = Alignment.CenterVertically
        ) {
            tabs.forEach { (label, key) ->
                val isSelected = currentTab == key
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.Center,
                    modifier = Modifier
                        .clickable { onSelectTab(key) }
                        .padding(horizontal = 8.dp, vertical = 4.dp)
                ) {
                    Text(
                        text = label,
                        fontSize = 12.sp,
                        fontWeight = if (isSelected) FontWeight.ExtraBold else FontWeight.Normal,
                        color = if (isSelected) OrbitCyan else MutedText
                    )
                    if (isSelected) {
                        Box(
                            modifier = Modifier
                                .padding(top = 4.dp)
                                .size(width = 16.dp, height = 2.dp)
                                .background(OrbitCyan)
                        )
                    }
                }
            }
        }
    }
}

// ─── 6. Orbit Specialized Feature Cards ──────────────────────────────────────

@Composable
fun OrbitPackageCard(
    title: String,
    subtitle: String,
    price: String,
    features: List<String>,
    onSelect: () -> Unit,
    isPopular: Boolean = false
) {
    OrbitGlassCard(borderColor = if (isPopular) OrbitPurple else OrbitBorder) {
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Column {
                Text(title, fontSize = 22.sp, fontWeight = FontWeight.Bold, color = Color.White)
                if (isPopular) {
                    Surface(color = OrbitPurple.copy(alpha = 0.2f), shape = RoundedCornerShape(4.dp)) {
                        Text("MOST POPULAR", color = OrbitPurple, fontSize = 10.sp, fontWeight = FontWeight.Bold, modifier = Modifier.padding(horizontal = 6.dp, vertical = 2.dp))
                    }
                }
            }
            Text(price, fontSize = 24.sp, fontWeight = FontWeight.Black, color = if (isPopular) OrbitPurple else OrbitCyan)
        }
        Text(subtitle, color = MutedText, fontSize = 13.sp, modifier = Modifier.padding(vertical = 12.dp))

        Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            features.forEach { feature ->
                Text("✓ $feature", color = Color.White, fontSize = 14.sp)
            }
        }

        Spacer(modifier = Modifier.height(20.dp))
        OrbitGradientButton(text = "Book $title Package", onClick = onSelect, modifier = Modifier.fillMaxWidth())
    }
}

@Composable
fun OrbitStatusTimeline(
    currentStage: Int,
    stages: List<String>
) {
    Column(verticalArrangement = Arrangement.spacedBy(10.dp)) {
        stages.forEachIndexed { index, stage ->
            Row(verticalAlignment = Alignment.CenterVertically) {
                if (index < currentStage) {
                    Text("✓ ", color = OrbitCyan, fontWeight = FontWeight.Bold)
                    Text(stage, color = Color.White, fontSize = 14.sp)
                } else if (index == currentStage) {
                    Text("• ", color = OrbitCyan, fontWeight = FontWeight.Bold)
                    Text("$stage (Active)", color = OrbitCyan, fontSize = 14.sp, fontWeight = FontWeight.Bold)
                } else {
                    Text("◦ ", color = MutedText)
                    Text("$stage (Pending)", color = MutedText, fontSize = 14.sp)
                }
            }
        }
    }
}

@Composable
fun OrbitWalletCard(
    balance: String,
    pending: String,
    onWithdraw: () -> Unit
) {
    OrbitGlassCard(borderColor = OrbitCyan) {
        Text("Available Balance", color = MutedText, fontSize = 13.sp)
        Text(balance, fontSize = 36.sp, fontWeight = FontWeight.Black, color = Color.White, modifier = Modifier.padding(vertical = 4.dp))
        Text("Pending Clearance: $pending", color = OrbitCyan, fontSize = 12.sp, fontWeight = FontWeight.Bold)
        Spacer(modifier = Modifier.height(16.dp))
        OrbitGradientButton(text = "Initiate Instant Payout", onClick = onWithdraw, modifier = Modifier.fillMaxWidth())
    }
}

@Composable
fun OrbitLoader() {
    Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
        CircularProgressIndicator(color = OrbitCyan, strokeWidth = 3.dp)
    }
}

@Composable
fun OrbitSkeleton() {
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .height(100.dp)
            .clip(RoundedCornerShape(16.dp))
            .background(SpaceNavyLight)
            .border(1.dp, OrbitBorder, RoundedCornerShape(16.dp))
    )
}
